pkg/autodiscovery/utils_autodiscovery.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build kubelet
+
+package autodiscovery
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
+)
+
+// GetAutodiscoveryErrors fetches AD errors from the kubelet ConfigProvider
+func (ac *AutoConfig) GetAutodiscoveryErrors(cpName string) map[string]map[string]bool {
+	for _, pd := range ac.providers {
+		if pd.provider.String() == cpName {
+			if cpName == names.Kubernetes {
+				return pd.provider.(*providers.KubeletConfigProvider).Errors
+			}
+		}
+	}
+	return nil
+}
```

Comment:
I think we can implement something more generic and future proof here:
we can extend the [ConfigProvider](https://github.com/DataDog/datadog-agent/blob/master/pkg/autodiscovery/providers/providers.go#L50) interface here and add a new method `ConfigErrors()` that can be implemented by any config provider. For the moment it's okay that only kubelet implements it, but more providers can implement it later on (they can just return an empty error list/map atm).
Like this, we can do something similar to:
```golang
	for _, pd := range ac.providers {
		errors[pd.provider.String()] = pd.provider.ConfigErrors()
	}
```